rpc/core/events.go

```diff
@@ -18,6 +18,50 @@ const (
 	maxQueryLength = 512
 )
 
+// Poll for new ABCI events.
+func Events(ctx *rpctypes.Context, query, maxWaitTime string) (event *ctypes.ResultEvent, err error) {
+	wt, err := time.ParseDuration(maxWaitTime)
+	if err != nil {
+		err = fmt.Errorf("failed to parse maxWaitTime: %w", err)
+		return
+	}
+
+	q, err := tmquery.New(query)
+	if err != nil {
+		err = fmt.Errorf("failed to parse query: %w", err)
+		return
+	}
+
+	const constMinWaitTime = 1 * time.Second
+	const constMaxWaitTime = 30 * time.Second
+	if wt < constMinWaitTime {
+		wt = constMinWaitTime
+	} else if wt > constMaxWaitTime {
+		wt = constMaxWaitTime
+	}
+
+	subCtx, cancel := context.WithTimeout(ctx.Context(), wt)
+	defer cancel()
+
+	addr := ctx.RemoteAddr()
+	sub, err := env.EventBus.Subscribe(subCtx, addr, q, env.Config.SubscriptionBufferSize)
+	if err != nil {
+		return
+	}
+	defer env.EventBus.Unsubscribe(context.Background(), addr, q)
```

Comment:
```suggestion
	defer func() {
		_ = env.EventBus.Unsubscribe(subCtx, addr, q)
	}()
```
We should use `subCtx` instead of `context.Background()` here? Also ideally we would log this error (at least at debug level) but we don't have a logger available to us here, for now we could explicitly discard it though that's far from ideal

Reply:
this error doesn't really matter much. we're just unsubscribing from the events at the pubsub layer.